### PR TITLE
shorten links

### LINKED ISSUE #50: Shorten links
- [x] use comic title instead of comic_id
	- [x] have a way to encode the title properly to be a valid url. (certain titles be be invalid in this case, should be alright)
- [x] use chapter number instead of chapter_id
- [x] make (user_id, title) unique in the comics table, instead of (title) only
- [ ] limit title length
- [ ] adhere to REST conventions in the APIs that uses slugs

### PR DESCRIPTION
**Pull request by mohad12211**
fixes #50 
- [x] use comic title instead of comic_id
	- [x] have a way to encode the title properly to be a valid url. (certain titles be be invalid in this case, should be alright)
- [x] use chapter number instead of chapter_id
- [x] make (user_id, title) unique in the comics table, instead of (title) only

### CODE REVIEW

File: client/src/routes/[username]/[comic_slug]/+page.svelte [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+<div class="comic-container">
+    <div class="comic">
+        <h1>{comic.title}</h1>
+        {#if comic.description}
+            <div>{comic.description}</div>
+        {/if}
+    </div>
+    <span>chapters:</span>
+    {#if $currentUser && $currentUser.id == comic.author.id}
+        <a href={`/${comic.author.username}/${comic.slug}/new-chapter`}
```
Comment (mohad12211) [inline client/src/routes/[username]/[comic_slug]/+page.svelte:61]: git is being stupid a bit about this, this was the only line changed

File: src/comics/chapters/routes.rs [hunk 2]
```diff
@@ -62,6 +63,10 @@ pub fn chapters_router() -> Router<AppState> {
         .route("/:comic_id/chapters", get(get_chapters))
         .route("/chapters/:chapter_id", delete(delete_chapter))
         .route("/chapters/:chapter_id", put(update_chapter))
+        .route(
+            "/chapters/by_slug/:username/:slug/:chapter_number/",
```
Comment (BKSalman) [inline src/comics/chapters/routes.rs:67]: I think it's better if it's like `/api/v1/comics/by_slug/:comic_slug/chapters/:chapter_number` no?

I think this is more consistent with the rest conventions and also makes it less prone to overlapping stuff

Comment (mohad12211) [reply src/comics/chapters/routes.rs:67]: What about `:username`?

Comment (BKSalman) [reply src/comics/chapters/routes.rs:67]: yeah, idk, we have 3 options, either stick it at the end like `/api/v1/comics/by_slug/:comic_slug/chapters/:chapter_number/:username` or `/api/v1/comics/by_slug/:comic_slug/chapters/:chapter_number/users/:username` or `/api/v1/users/:username/comics/by_slug/:comic_slug/chapters/:chapter_number`

maybe last one is the best?

File: src/comics/routes.rs [hunk 3]
```diff
@@ -38,6 +39,7 @@ pub fn comics_router() -> Router<AppState> {
         .route("/:comic_id", put(update_comic))
         .route("/:comic_id", delete(delete_comic))
         .route("/:comic_id", get(get_comic))
+        .route("/by_slug/:slug/:username", get(get_comic_by_slug))
```
Comment (mohad12211) [inline src/comics/routes.rs:42]: if we are gonna go by
`/api/v1/users/:username/comics/by_slug/:comic_slug/chapters/:chapter_number`
should this be
`/api/v1/users/:username/comics/by_slug/:comic_slug`
?

Comment (BKSalman) [reply src/comics/routes.rs:42]: Yes, to keep consistency in the API

Even tho in code, it will be a bit meh, but the API is what matters

Comment (mohad12211) [reply src/comics/routes.rs:42]: later :tm: 
if code is good and logical, refactoring will be easy later.

**Review by BKSalman — APPROVED** [review]
later :tm: + LGTM LGTM

**Pull request merged**